desmod/workspacesync/s3.py

```diff
@@ -0,0 +1,59 @@
+from concurrent.futures import ThreadPoolExecutor
+
+import boto3
+import getpass
+import os
+
+
+class S3Sync(object):
```

Comment:
Even without using classes, each synchronization backend module could register its `sync()` function. Lots of ways to skin this. As long as these syncronization backends are all driven by configuration and not by user-facing API, then we retain lots of license to revisit this detail.

I do think there is at least one compelling case for having a class: extract and check the configuration *before* simulation.